### PR TITLE
Adding a configurable config location through an env var.

### LINKED ISSUE #764: Configurable config location
I'm working on packaging Lemmy for Alpine Linux. Since it would become part of a regular distribution, the config file can not be in `/config/` as packages are not allowed to write to that directory. Please allow specifying an alternative config location, for example with a simple command-line switch (e.g. `/usr/bin/lemmy_server -c /etc/lemmy/config.hjson`), or look in `/etc` by default.

### PR DESCRIPTION
**Pull request by dessalines**
- Its `LEMMY_CONFIG_LOCATION`
- Fixes #764

### CODE REVIEW

File: server/lemmy_utils/src/settings.rs [hunk 1]
```diff
@@ -115,12 +115,16 @@ impl Settings {
     format!("{}/api/v1", self.hostname)
   }
 
+  pub fn get_config_location() -> String {
+    env::var("LEMMY_CONFIG_LOCATION").unwrap_or("config/config.hjson".to_string())
```
Comment (Nutomic) [inline server/lemmy_utils/src/settings.rs:119]: I would keep the variable CONFIG_FILE, instead of putting it here inline.

**Comment by Nutomic** [comment]
You should probably make an env var for CONFIG_FILE_DEFAULTS as well.

File: server/lemmy_utils/src/settings.rs (hunk 1 above)
Comment (dessalines) [reply server/lemmy_utils/src/settings.rs:119]: Ah okay, Ya I can do that.

**Comment by dessalines** [comment]
> You should probably make an env var for CONFIG_FILE_DEFAULTS as well.

This one I don't get, mainly because the defaults are overridden whenever you provide a custom one anyway. So it only makes sense to make one.

**Comment by Nutomic** [comment]
It just seems cleaner to have one file with the defaults, which can be updated whenever, and a separate file only which only contains things that are changed from the defaults. But at least for now this should be enough.

**Pull request merged**